include/capo/source.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+#include <capo/types.hpp>
+#include <capo/utils/id.hpp>
+
+namespace capo {
+class Sound;
+class Instance;
+
+///
+/// \brief Lightweight handle to an audio source in 3D space; use Instance to create
+///
+/// Can bind to Sound clips, play/pause/stop, loop, etc.
+///
+class Source {
+  public:
+	Source() = default;
+	static Source const blank;
+
+	bool valid() const noexcept { return m_instance && m_handle > 0; }
+	bool bind(Sound const& sound);
```

Comment:
Don't you have something like a `Result<void, void>` type in ktl we could use here? bool isn't really indicative of what the return value means

Reply:
I think that would be overkill. 😛 The inline docs on the functions will specify what the return value indicates (aka `true` if successful or if inactive with `valid_even_inactive_v` being `true`).